### ISSUE TITLE
Move EWMH hook to its own package

### ISSUE THREAD

**Issue by IvanMalison** [travitch/taffybar#292]
Our EWMH hooks module should really be in its own libarary, that doesn't require all of the dependencies that taffybar does. The fact that it isn't means that stack builds for xmonad that include this have to include the mountain of extra-deps that taffybar currently requires.

Once https://github.com/haskell/cabal/issues/4206 is completed we should separate the hook out into its own package.

**Comment by mimi1vx** [comment]
haskell/cabal#4206 is pretty bad idea ..

**Comment by IvanMalison** [comment]
@mimi1vx Why do you think that?

**Comment by mimi1vx** [comment]
@IvanMalison is good for developer,  but for packager is this nightmare.

**Comment by IvanMalison** [comment]
I think the best approach might be to move this into xmonad-contrib.

**Comment by LeifW** [comment]
As long as that doesn't add a dep on taffybar. What would its deps be?. Otherwise, there's xmonad-extras - which is for xmonad extensions with wacky deps - it uses configure flags or other mechanisms to conditionally build modules that depend on larger / more uncommon things. Not sure how well-maintained it is these days, but I still use it (for the volume and mpd stuff, mainly).
Though, I don't see what's wrong with just splitting this out into its own separate stand-alone lib?

**Comment by IvanMalison** [comment]
@LeifW I don't think it has a dependency on anything within taffybar itself see https://github.com/taffybar/taffybar/blob/58bc270c9d45329a5cdd9a8b09fd0ad778fbae31/src/System/Taffybar/Support/PagerHints.hs#L39

Seems like this would not require adding any deps to xmonad-contrib.

**Comment by LeifW** [comment]
There seems to be a few other libs on hackage that are built on XMonad - I was picturing it could be like one of those. Though this seems quite small, and there's already stuff for dzen and xmobar in xmonad-contrib's XMonad.Hooks.DynamicLog.

**Comment by nomeata** [comment]
I see https://hackage.haskell.org/package/xmonad-contrib-0.17.0/docs/XMonad-Hooks-TaffybarPagerHints.html now – I guess this means this issue can be closed?

#526 is about removing the code from here.

**Comment by IvanMalison** [comment]
@nomeata yes, My plan was to close the issue once a new version of taffybar has actually been released.
